### PR TITLE
Document Bounce = 1.0 not being sufficient for infinite energy conservation

### LINKED ISSUE #82945: Rigidbody lose bounce enegy with bounce 1
### Godot version

4.1.1

### System information

Godot v4.1.1.stable unknown - Manjaro Linux #1 SMP PREEMPT_DYNAMIC Wed Sep 13 14:10:57 UTC 2023 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1070 (nvidia; 535.104.05) - Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 Threads)

### Issue description

Well i created simple test project to test bounces and found that physics calculations in my opinion work not right. 
I am not sure is it a bug but i almost sure that exist calculation mistake which in my opinion too big. And due to it Rigidbody loose energy too fast than it have to be in case if used bounce property = 1. In ideal situation i expect that Rigidbody have to bounce endless to same height. I attached simple project which illustrate issue.

### Steps to reproduce

1) load and start attached project.
2) wait 5-10 seconds 
3) See that ball loose height quite fast

### Minimal reproduction project

[sidegame1.zip](https://github.com/godotengine/godot/files/12836981/sidegame1.zip)

### PR DESCRIPTION
**Pull request by Calinou**
- This closes https://github.com/godotengine/godot/issues/82945.

### CODE REVIEW

File: doc/classes/PhysicalBone3D.xml [hunk 1]
```diff
@@ -61,6 +61,7 @@
 		</member>
 		<member name="bounce" type="float" setter="set_bounce" getter="get_bounce" default="0.0">
 			The body's bounciness. Values range from [code]0[/code] (no bounce) to [code]1[/code] (full bounciness).
+			[b]Note:[/b] Even with [member bounce] set to [code]1.0[/code], some energy will be lost over time due to linear and angular damping. To have a [PhysicalBone3D] that preserves all its energy over time, set [member bounce] to [code]1.0[/code], [member linear_damp_mode] to [constant DAMP_MODE_REPLACE], [member linear_damp] to [code]0.0[/code], [member angular_damp_mode] to [constant DAMP_MODE_REPLACE] and [member angular_damp] to [code]0.0[/code].
```
Comment (AThousandShips) [inline doc/classes/PhysicalBone3D.xml:64]:
```suggestion
			[b]Note:[/b] Even with [member bounce] set to [code]1.0[/code], some energy will be lost over time due to linear and angular damping. To have a [PhysicalBone3D] that preserves all its energy over time, set [member bounce] to [code]1.0[/code], [member linear_damp_mode] to [constant DAMP_MODE_REPLACE], [member linear_damp] to [code]0.0[/code], [member angular_damp_mode] to [constant DAMP_MODE_REPLACE], and [member angular_damp] to [code]0.0[/code].
```

File: doc/classes/PhysicsMaterial.xml [hunk 2]
```diff
@@ -14,6 +14,7 @@
 		</member>
 		<member name="bounce" type="float" setter="set_bounce" getter="get_bounce" default="0.0">
 			The body's bounciness. Values range from [code]0[/code] (no bounce) to [code]1[/code] (full bounciness).
+			[b]Note:[/b] Even with [member bounce] set to [code]1.0[/code], some energy will be lost over time due to linear and angular damping. To have a [PhysicsBody3D] that preserves all its energy over time, set [member bounce] to [code]1.0[/code], the body's linear damp mode to Replace (if applicable), its linear damp to [code]0.0[/code], its angular damp mode to Replace (if applicable) and its angular damp to [code]0.0[/code].
```
Comment (AThousandShips) [inline doc/classes/PhysicsMaterial.xml:17]:
```suggestion
			[b]Note:[/b] Even with [member bounce] set to [code]1.0[/code], some energy will be lost over time due to linear and angular damping. To have a [PhysicsBody3D] that preserves all its energy over time, set [member bounce] to [code]1.0[/code], the body's linear damp mode to Replace (if applicable), its linear damp to [code]0.0[/code], its angular damp mode to Replace (if applicable), and its angular damp to [code]0.0[/code].
```

File: doc/classes/PhysicsMaterial.xml [hunk 3]
```diff
@@ -14,6 +14,7 @@
 		</member>
 		<member name="bounce" type="float" setter="set_bounce" getter="get_bounce" default="0.0">
 			The body's bounciness. Values range from [code]0[/code] (no bounce) to [code]1[/code] (full bounciness).
+			[b]Note:[/b] Even with [member bounce] set to [code]1.0[/code], some energy will be lost over time due to linear and angular damping. To have a [PhysicsBody3D] that preserves all its energy over time, set [member bounce] to [code]1.0[/code], the body's linear damp mode to Replace (if applicable), its linear damp to [code]0.0[/code], its angular damp mode to Replace (if applicable), and its angular damp to [code]0.0[/code].
```
Comment (Mickeon) [inline doc/classes/PhysicsMaterial.xml:17]: I gather `[constant RigidBody3D.DAMP_MODE_REPLACE]` would have made this note even more verbose than it already is, but at least the quotes:
```suggestion
			[b]Note:[/b] Even with [member bounce] set to [code]1.0[/code], some energy will be lost over time due to linear and angular damping. To have a [PhysicsBody3D] that preserves all its energy over time, set [member bounce] to [code]1.0[/code], the body's linear damp mode to "Replace" (if applicable), its linear damp to [code]0.0[/code], its angular damp mode to "Replace" (if applicable), and its angular damp to [code]0.0[/code].
```

**Review by Mickeon — APPROVED** [review]
This is good information, even if the note itself is _extremely_ detailed, perhaps too much.
I would tone these instructions down by a large margin but I won't cry myself to sleep about it.

File: doc/classes/PhysicsMaterial.xml (hunk 3 above)
Comment (Calinou) [reply doc/classes/PhysicsMaterial.xml]: I've used bold instead which is what we use to highlight menu options in the editor.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.
Cherry-picked for 4.1.4.

**Pull request merged**